alerts_consumer.go

```diff
@@ -202,24 +250,37 @@ func (c *AlertsConsumer) SendBatch(batch [][]byte, tag string) error {
 		}
 	}
 
+	var err error
 	retry := retrier.New(retrier.ExponentialBackoff(5, 50*time.Millisecond), nil)
-	retry.Run(func() error {
+
+	err = retry.Run(func() error {
 		lg.TraceD("sfx-add-datapoints", logger.M{"point-count": len(pts)})
 		return c.sfxSink.AddDatapoints(context.Background(), pts)
 	})
-	var err error
 	if err != nil && err.Error() == "invalid status code 400" { // internal buffer full in sfx
 		return kbc.PartialSendBatchError{ErrMessage: "failed to add datapoints: " + err.Error(), FailedMessages: batch}
 	}
 
-	retry = retrier.New(retrier.ExponentialBackoff(5, 50*time.Millisecond), nil)
-	retry.Run(func() error {
+	err = retry.Run(func() error {
 		lg.TraceD("sfx-events", logger.M{"point-count": len(evts)})
 		return c.sfxSink.AddEvents(context.Background(), evts)
 	})
 	if err != nil && err.Error() == "invalid status code 400" { // internal buffer full in sfx
 		return kbc.PartialSendBatchError{ErrMessage: "failed to add events: " + err.Error(), FailedMessages: batch}
 	}
 
+	err = retry.Run(func() error {
+		lg.Trace("cloudwatch-add-datapoints")
+		_, e := c.cwAPI.PutMetricData(&cloudwatch.PutMetricDataInput{
+			Namespace:  aws.String(cloudwatchNamespace),
+			MetricData: dats,
+		})
+		return e
+	})
+	// TODO: We could return that we failed to process this batch (as above) but for now just log
+	if err != nil {
+		lg.ErrorD("error-sending-to-cloudwatch", logger.M{"error": err.Error()})
```

Comment:
I wonder if it's worth surfacing this error through something similar to https://github.com/Clever/kinesis-alerts-consumer/blob/91a0c74b39274f75c229235b624c81aada43db80/kvconfig.yml#L8 rather than just logging it (no one reads those)

it's consuming it's own messages, but 🤷‍♂

Reply:
Makes sense, I'll do that.